### PR TITLE
support localization bilingual page build

### PR DESCRIPTION
**Pull request by OsmondJiang**

### CODE REVIEW

**Comment by OsmondJiang** [comment]
#3465

File: src/docfx/localization/LocalizationConvention.cs [hunk 1]
```diff
@@ -73,20 +84,24 @@ public static string GetLocalizationDocsetPath(string docsetPath, Config config,
...
+                        localizationDocsetPath = config.Localization.Mapping == LocalizationMapping.Repository
                             ? restorePath
                             : Path.Combine(restorePath, locale);
                         break;
                     }
                 case LocalizationMapping.Folder:
                     {
+                        if (config.Localization.Bilingual)
+                        {
+                            throw new NotSupportedException($"{config.Localization.Mapping} is not supporting bilingual build");
```
Comment (live1206) [inline src/docfx/localization/LocalizationConvention.cs]: test for this corner case?

File: src/docfx/localization/LocalizationConvention.cs [hunk 2]
```diff
@@ -21,38 +21,49 @@ internal static class LocalizationConvention
...
+        public static (string remote, string branch) GetLocalizationRepo(LocalizationConfig localizationConfig, string remote, string branch, string locale, string defaultLocale)
         {
-            if (localizationMapping != LocalizationMapping.Repository && localizationMapping != LocalizationMapping.RepositoryAndFolder)
+            if (localizationConfig == null)
             {
-                return remote;
+                return (remote, branch);
+            }
+
+            if (localizationConfig.Mapping != LocalizationMapping.Repository && localizationConfig.Mapping != LocalizationMapping.RepositoryAndFolder)
```
Comment (live1206) [inline src/docfx/localization/LocalizationConvention.cs]: make it shorter since it is just folder mapping here?
```csharp
if (localizationConfig.Mapping == LocalizationMapping.Folder)
```

Comment (OsmondJiang) [reply src/docfx/localization/LocalizationConvention.cs]: i want to use the write list, not black list in case we add more mappings later

---
In reply to: [227691489](https://github.com/dotnet/docfx/pull/3528#discussion_r227691489) [](ancestors = 227691489)

**Comment by andreasblueher** [comment]
Hey @OsmondJiang is there any place where I can see an example on how to use this bilingual implementation?

**Comment by OsmondJiang** [comment]
@andreasblueher , here is the spec: https://github.com/dotnet/docfx/pull/3547/files

and also most of the content/pages in docs.com has applied bilingual feature, like this page: https://docs.microsoft.com/ja-jp/azure/virtual-machines/windows/quick-create-powershell

image
